biotrainer/embedders/embedding_service.py

```diff
@@ -68,45 +75,146 @@ def compute_embeddings(self, sequence_file: str, output_dir: Path, protocol: Pro
         protein_sequences = {seq.id: str(seq.seq) for seq in sorted(read_FASTA(sequence_file),
                                                                     key=lambda seq: len(seq.seq),
                                                                     reverse=True)}
+
+        embeddings_file_path = self._do_embeddings_computation(protein_sequences, embeddings_file_path, use_reduced_embeddings)
+        
+        gc.collect()
+
+        return str(embeddings_file_path)
+    
+    def _do_embeddings_computation(self, protein_sequences: Dict[str, str], embeddings_file_path: Path,
+                          use_reduced_embeddings: bool) -> str:
+        """
+        Performs the embedding service for the given protein sequences.
+
+        Parameters:
+            protein_sequences (Dict[str, str]): A dictionary mapping sequence IDs to protein sequences.
+            embeddings_file_path (Path): The path where embeddings will be saved.
+            use_reduced_embeddings (bool): Indicates if reduced embeddings should be used.
+
+        Returns:
+            str: The path to the embeddings file.
+        """
         sequence_ids = list(protein_sequences.keys())
         embeddings = {}
         idx: int = 0
         last_save_id: int = 0
-        for embedding in tqdm(self._embedder.embed_many(protein_sequences.values()),
-                              total=len(protein_sequences.values())):
-            embeddings[sequence_ids[idx]] = embedding
-            idx += 1
-            if len(embeddings) > SAVE_AFTER_N_EMBEDDINGS:
-                if use_reduced_embeddings:
-                    embeddings = self._reduce_embeddings(embeddings, self._embedder)
-                last_save_id = self._save_embeddings(save_id=last_save_id, embeddings=embeddings,
-                                                     embeddings_file_path=embeddings_file_path)
-
-                # Manually clear to ensure that embeddings are deleted from RAM
-                del embeddings
-                embeddings = {}
-
+        start_time = time.time()
+        
+        embedding_iter = self._embedder.embed_many(protein_sequences.values())
+        total_sequences = len(protein_sequences.values())
+        
+        # Load the first sequence and calculate the initial max_embedding_fit
+        embeddings[sequence_ids[0]] = next(embedding_iter, None)
+        max_embedding_fit = self._max_embedding_fit(embeddings[sequence_ids[0]])
+        
+        if embeddings[sequence_ids[0]] is None:
+            logger.info(f"No embeddings found.")
+            return str(embeddings_file_path)
+        
+        logger.info(f"Embedding dimension: {embeddings[sequence_ids[idx]].shape[-1]}")
+        logger.info("If your dataset contains long reads, it may take more time to process the first few sequences.")
+        
+        with tqdm(total=total_sequences, initial=1, desc="Processing Sequences") as pbar:
```

Comment:
I just checked locally, and the processing bar is not shown yet for the first embedding calculation. Could you move the whole line to line 108, then it should be shown. You have to double check if the initial value is still correct (0 or 1?) and probably have to call `pbar.update(1)` once manually after the line 108. But I think the improved user feedback is worth this small overhead.

Additionally, I would suggest to use `desc="Computing embeddings"` to be clearer about the description (but adding the description is a good idea anyway, thanks!)